src/composables/transactionData.ts

```diff
@@ -171,32 +181,37 @@ export function useTransactionData({
       ),
   );
 
+  /**
+   * Amount and fee calculated based on the direction.
+   */
+  const amountTotal = computed(
+    (): number => (transaction.value) ? getTxAmountTotal(transaction.value, direction.value) : 0,
+  );
+
   /**
    * List of assets used within the transaction.
    * Contains more than one item if the transaction is for example a swapping event.
    */
   const transactionAssets = computed((): ITokenResolved[] => {
-    if (!activeTransaction.value?.tx) {
+    if (!transaction.value?.tx) {
       return [];
     }
 
     let convertToCoin = false;
 
-    const { protocol = PROTOCOLS.aeternity } = activeTransaction.value;
-    const adapter = ProtocolAdapterFactory.getAdapter(protocol);
-    const protocolTokens = getProtocolAvailableTokens(protocol);
+    const adapter = ProtocolAdapterFactory.getAdapter(protocol.value);
+    const protocolTokens = getProtocolAvailableTokens(protocol.value);
 
     // TODO move AE specific logic to adapter and store resolved data in the transactions
-    if (protocol === PROTOCOLS.aeternity) {
+    if (protocol.value === PROTOCOLS.aeternity) {
       // AE DEX and wrapped AE (WAE)
       if (
-        innerTx.value?.function
+        isDex.value
+        && txFunctionParsed.value
         && (!isDexAllowance.value || showDetailedAllowanceInfo)
       ) {
-        const functionName = camelCase(innerTx.value.function) as TxFunctionParsed;
-        const functionResolver = getTransactionTokenInfoResolver(functionName);
-
-        if (functionResolver && isDex.value) {
```

Comment:
What do you think @CedrikNikita about moving `isDex` from this place to the if test one level above?